matrix-sdk/src/main/java/org/matrix/androidsdk/data/store/MXFileStore.java

```diff
@@ -2346,7 +2350,17 @@ private Object readObject(String description, File file) {
         Object object = null;
         try {
             FileInputStream fis = new FileInputStream(file);
-            GZIPInputStream gz = new GZIPInputStream(fis);
+            InputStream cis = CompatUtil.createCipherInputStream(fis, mContext);
```

Comment:
It would not be a security measure if when something wrong happen you just drop it out. If you fail to establish a HTTPS (reps. SSH) session for instance, it is not expected to fallback to HTTP (resp. TELNET).